internals/overlord/servstate/handlers.go

```diff
@@ -357,32 +357,36 @@ func (s *serviceData) startInternal() error {
 	s.cmd = exec.Command(args[0], args[1:]...)
 	s.cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
 
-	// Copy environment to avoid updating original.
-	var environment map[string]string
-	if s.workload != nil && len(s.workload.Environment) > 0 {
-		environment = maps.Clone(s.workload.Environment)
-	} else if len(s.config.Environment) > 0 {
-		environment = maps.Clone(s.config.Environment)
-	} else {
-		environment = make(map[string]string)
+	environment := make(map[string]string)
+	if s.workload != nil {
+		maps.Copy(environment, s.workload.Environment)
 	}
+	maps.Copy(environment, s.config.Environment)
```

Comment:
Is it the intended behavior to merge the two environment maps in case both the one from the workload and the config are non-empty?